storage/src/linear/filebacked.rs

```diff
@@ -50,15 +63,31 @@ impl ReadableStorage for FileBacked {
             .expect("poisoned lock")
             .seek(std::io::SeekFrom::End(0))
     }
+
+    fn read_cached_node(&self, addr: LinearAddress) -> Option<Arc<Node>> {
+        let mut guard = self.cache.lock().expect("poisoned lock");
```

Comment:
My reasoning, right or wrong, is that this lock is very short lived and unlikely to contend with other readers, so the overhead of RwLock isn't worth it.

In general, I prefer the simplicity and speed of a Mutex over a RwLock since it can often be machine optimized better than an RwLock if i think it's not going to contend much. We can always change it later if that proves to be false.